pkg/registry/service/ip_allocator.go

```diff
@@ -1,250 +0,0 @@
-/*
-Copyright 2014 The Kubernetes Authors All rights reserved.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package service
-
-import (
-	"fmt"
-	math_rand "math/rand"
-	"net"
-	"sync"
-	"time"
-
-	"github.com/golang/glog"
-)
-
-type ipAllocator struct {
-	lock sync.Mutex // protects 'used'
-
-	subnet         net.IPNet
-	ipSpaceSize    int64 // Size of subnet, or -1 if it does not fit in an int64
-	used           ipAddrSet
-	randomAttempts int
-
-	random *math_rand.Rand
-}
-
-type ipAddrSet struct {
-	// We are pretty severely restricted in the types of things we can use as a key
-	ips map[string]bool
-}
-
-func (s *ipAddrSet) Init() {
-	s.ips = map[string]bool{}
-}
-
-// Gets the number of IPs in the set
-func (s *ipAddrSet) Size() int {
-	return len(s.ips)
-}
-
-// Tests whether the set holds a given IP
-func (s *ipAddrSet) Contains(ip net.IP) bool {
-	key := ip.String()
-	exists := s.ips[key]
-	return exists
-}
-
-// Adds to the ipAddrSet; returns true iff it was added (was not already in set)
-func (s *ipAddrSet) Add(ip net.IP) bool {
-	key := ip.String()
-	exists := s.ips[key]
-	if exists {
-		return false
-	}
-	s.ips[key] = true
-	return true
-}
-
-// Removes from the ipAddrSet; returns true iff it was removed (was already in set)
-func (s *ipAddrSet) Remove(ip net.IP) bool {
-	key := ip.String()
-	exists := s.ips[key]
-	if !exists {
-		return false
-	}
-	delete(s.ips, key)
-	// TODO: We probably should add this IP to an 'embargo' list for a limited amount of time
-
-	return true
-}
-
-// The smallest number of IPs we accept.
-const minIPSpace = 8
-
-// newIPAllocator creates and intializes a new ipAllocator object.
-func newIPAllocator(subnet *net.IPNet) *ipAllocator {
-	if subnet == nil || subnet.IP == nil || subnet.Mask == nil {
-		return nil
-	}
-
-	seed := time.Now().UTC().UnixNano()
-	r := math_rand.New(math_rand.NewSource(seed))
-
-	ipSpaceSize := int64(-1)
-	ones, bits := subnet.Mask.Size()
-	if (bits - ones) < 63 {
-		ipSpaceSize = int64(1) << uint(bits-ones)
-
-		if ipSpaceSize < minIPSpace {
-			glog.Errorf("IPAllocator requires at least %d IPs", minIPSpace)
-			return nil
-		}
-	}
-
-	ipa := &ipAllocator{
-		subnet:         *subnet,
-		ipSpaceSize:    ipSpaceSize,
-		random:         r,
-		randomAttempts: 1000,
-	}
-	ipa.used.Init()
-
-	network := make(net.IP, len(subnet.IP), len(subnet.IP))
```

Comment:
You dropped this logic - it seems that it is possible to get allocated the network or broadcast address now.